ingress/proxy.go

```diff
@@ -0,0 +1,16 @@
+package ingress
+
+import "github.com/sky-uk/feed/controller"
+
+// Proxy proxies traffic from external clients to the internal services in Kubernetes.
+type Proxy interface {
```

Comment:
Well we actually had `type Proxy controller.Updater`. I think it's nicer to make it explicitly for proxies rather than reuse updater, otherwise it gets pretty confusing having updaters calling updaters.

But the bool return value on Update doesn't make sense for the dns updater, so the interfaces don't actually match currently.